sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -160,8 +163,6 @@ private Tree convertElementToSlangAST(PsiElement element, TreeMetaData metaData)
     } else if (element instanceof KtDestructuringDeclarationEntry || isSimpleStringLiteralEntry(element)) {
       // To differentiate between the native trees of complex string template entries, we add the string value to the native kind
       return createNativeTree(metaData, new KotlinNativeKind(element, element.getText()), element);
-    } else if (element instanceof KtParameter) {
```

Comment:
We can keep that part I believe. That way for normal loops (Ex: `for (a in container) {a}`) we keep information about `ParameterTree` for the variable. And we can add a test for this case